src/chigame/games/views.py

```diff
@@ -148,20 +148,35 @@ class TournamentCreateView(CreateView):
         "rules",
         "draw_rules",
         "num_winner",
-        "matches",
         "players",
     ]
-    # Note: "winner" is not included in the fields because it is not
-    # supposed to be set by the user. It will be set automatically
-    # when the tournament is over.
-    # Note: we may remove the "matches" field later for the same reason,
-    # but we keep it for now because it is convenient for testing.
+
+    def form_valid(self, form):
+        user = self.request.user
+        if user.tokens < 1:
+            # Raise a PermissionDenied exception to show the forbidden page
+            raise PermissionDenied("You do not have enough tokens to create a tournament.")
+
+        # Deduct a token from the user's account
+        user.tokens -= 1
+        user.save()
+
+        # Save the form instance but don't commit to the database yet
+        tournament = form.save(commit=False)
+        tournament.created_by = user
+        tournament.save()
+
+        # Add the creator to the players list
+        tournament.players.add(user)
```

Comment:
This line seems to be making it so that only the creator is added to the players list. 
`players = form.cleaned_data['players']
 tournament.players.add(*players)`
 will fix this.